lib/usd/ui/layerEditor/layerTreeItem.cpp

```diff
@@ -351,7 +351,9 @@ void LayerTreeItem::saveEditsNoPrompt()
         if (!isSessionLayer())
             saveAnonymousLayer();
     } else {
-        layer()->Save();
+        PXR_NS::SdfFileFormat::FileFormatArguments args;
+        args["format"] = MayaUsd::utils::usdFormatArgOption();
+        layer()->Export(layer()->GetRealPath(), "", args);
```

Comment:
Seems okay to me, but I'm wondering what are the implications of calling Export() vs Save()? @ppt-adsk Do you have any comments about this?

Reply:
We use Export in many other place to save when we want to control the output format.

Reply:
As per the docs, Save() saves to the previously-remembered file name, and fails if there is none.  Export() saves a copy of the layer, and does not affect the previously-remembered file name.  There is one important point: Save() will clear the layer's dirty state (i.e. SdfLayer::IsDirty() will return false), whereas Export() will not, so this change is introducing an important difference.

Reply:
Export() calls _WriteToFile() which calls _MarkCurrentStateAsClean() if the filename did not change, which is our case since we're exporting to the same file (we pass GetRelPath()).

The differences I can see are:
- Save() does nothing if the layer is not dirty. We could add a call to IsDirty().
- Set modification time, which I don't think we care about.

Things that look different but are the same in the end:
- Save() check IsMuted() and IsAnonymous() directly, but Export() ends-up doing the same check because _WriteToFile() calls PermissionToSave(0 when the filename is the same and that does the same checks.
- Save(0 clears hints, but the hints are not used once the dirty state is reset.

Reply:
... unfortunately, there are two other places in the code where we call Save() directly on the layer... so they will not be aware of the changed file format. I'm taking a deeper look but we will either have to modify those too or abandon the idea of switching file formats for already-saved layers.